spacepy/pybats/gitm.py

```diff
@@ -23,103 +25,212 @@ class GitmBin(PbData):
-
-            # Get code version:
-            self.attrs['version']=readarray(infile,floattype,inttype)[0]
-
-            # Read grid size information.  Because these three values
-            # are written at the same time, they get "packed" together and
-            # must be read together.
-            header_fields_dtype=np.dtype([
-                ('nLon',np.int32), ('nLat',np.int32), ('nAlt',np.int32)])
-            header_fields_dtype.newbyteorder(EndChar)
-            self.attrs['nLon'], self.attrs['nLat'], self.attrs['nAlt'] = \
-                readarray(infile,header_fields_dtype,inttype)[0]
-                
-            # Read number of variables:
-            self.attrs['nVars'] = readarray(infile,inttype,inttype)[0]
-
-            # Collect variable names; decode into usable strings.
-            var=[]
-            for i in range(self.attrs['nVars']):
-                var.append(readarray(infile,str,inttype).decode('utf-8'))
-
-            # Extract time, stored as 7 consecutive integers.
-            (yy,mm,dd,hh,mn,ss,ms)=readarray(infile,inttype,inttype)
-            self['time']=dt.datetime(yy,mm,dd,hh,mn,ss,ms//1000)
-
-            # Read the rest of the data.
-            nTotal=self.attrs['nLon']*self.attrs['nLat']*self.attrs['nAlt']
-            for val in var:
-                # Trim variable names.
-                v=sub(r'\[|\]', '', val).strip()
-                self[v] = readarray(infile,floattype,inttype)
-                # Reshape arrays, note that ordering in file is Fortran-like.
-                self[v]=self[v].reshape( 
-                    (self.attrs['nLon'],self.attrs['nLat'],self.attrs['nAlt']),
-                    order='F')
-                # Reduce dimensionality:
-                self[v] = np.squeeze(self[v])
+        isFirstTime = True
+        for iFile, filename in enumerate(self.attrs['files']):
+            with open(filename, 'rb') as file:
+                # On the first try, we may fail because of wrong-endianess.
+                # If that is the case, swap that endian and try again.
+                endChar = '>'
+                rawRecLen = file.read(4)
+                rec_len = (unpack(endChar + 'l', rawRecLen))[0]
+                if rec_len > 10000 or rec_len < 0:
+                    # Ridiculous record length implies wrong endian.
+                    endChar='<'
+
+                # Only read time if _read_header has not (it reads the 1st time)
+                if not isFirstTime:
+                    # Extract time, stored as 7 consecutive integers.
+                    # time is placed after variable names, so we skip:
+                    # 52(grid etc.) + ( 40[nvars] + 8 [head/foot] ) *nVarsTotal
+                    file.seek(52 + 48 * self.attrs['nVarsTotal'])
+                    (yy,mm,dd,hh,mn,ss,ms,startstop)=unpack(endChar+'llllllli',file.read(32))
+                    self.attrs['time'][iFile]=dt.datetime(yy,mm,dd,hh,mn,ss,ms * 1000)
```

Comment:
Using this as an example (relevant elsewhere) but [PEP-8 whitespace formatting](https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements) would be helpful here...